drivers/include/periph/gpio.h

```diff
@@ -10,6 +10,30 @@
  * @defgroup    drivers_periph_gpio GPIO
  * @ingroup     drivers_periph
  * @brief       Low-level GPIO peripheral driver
+ * 
+ * This is a basic GPIO (General-purpose input/output) interface to allow 
+ * platform independent access to a MCU's input/output pins. This interface is 
+ * intentionally designed to be as simple as possible, to allow for easy 
+ * implementation and maximal portability.
+ *
+ * The interface provides the user with capabilities to initialize a pin as output-, 
+ * input- and interrupt pin. With the API you can basically set/clear/toggle the 
+ * digital signal at the hardware pin when in output mode. Configured as input you can 
+ * read a digital value that is applied to the pin externally. When initializing an
+ * interrupt pin, you can register a callback function that is executed in interrupt
+ * context once the interrupt condition applies to the pin. Usually you can react to 
+ * positive or negative signal flanks (or either both).
+ *
+ * In addition the API provides to set standard input/output circuit modes such as 
+ * e.g. internal push-pull configurations.
+ *
+ * A hardware pin is described by its port-pin tupel. Do select a pin, the macro
+ * `GPIO_PIN(port, pin)` should be used. For example: If your platform has a pin PA01,
+ * it will be port=0 and pin=1. Usually the specific driver implementation
```

Comment:
maybe use PB22 as example, because you mean A==0 and 01==1, but the A and the leading zero are ambiguous